src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -0,0 +1,408 @@
+                        // Also don't need to worry other changes since the member is removed
+                        originalMemberEditor.RemoveNode(originalMemberEditor.Generator.GetDeclaration(syntax));
+                    }
+                    else
+                    {
+                        if (analysisResult.ChangeOriginalToNonStatic || analysisResult.ChangeOriginalToPublic)
+                        {
+                            ChangeMemberToPublicAndNonStatic(
+                                codeGenerationService, originalMemberEditor,
+                                syntax, analysisResult.Member);
+                        }
+                    }
+                }
+            }
+
+            return solutionEditor.GetChangedSolution();
+        }
+
+        private void ChangeMemberToPublicAndNonStatic(
+            ICodeGenerationService codeGenerationService,
+            DocumentEditor editor,
+            SyntaxNode memberSyntax,
+            ISymbol member)
+        {
+            var modifiers = DeclarationModifiers.From(member).WithIsStatic(false);
+            if (member is IEventSymbol eventSymbol)
+            {
+                ChangeEventToPublicAndNonStatic(
+                    codeGenerationService,
+                    editor,
+                    eventSymbol,
+                    memberSyntax,
+                    modifiers);
+            }
+            else
+            {
+                editor.SetAccessibility(memberSyntax, Accessibility.Public);
+                editor.SetModifiers(memberSyntax, modifiers);
+            }
+        }
+
+        private void ChangeEventToPublicAndNonStatic(
+            ICodeGenerationService codeGenerationService,
+            DocumentEditor editor,
+            IEventSymbol eventSymbol,
+            SyntaxNode memberSyntax,
+            DeclarationModifiers modifiers)
+        {
+            var declaration = editor.Generator.GetDeclaration(memberSyntax);
+            if (declaration.Equals(memberSyntax))
+            {
+                if ((eventSymbol.AddMethod != null && !eventSymbol.AddMethod.IsImplicitlyDeclared) ||
+                    (eventSymbol.RemoveMethod != null && !eventSymbol.RemoveMethod.IsImplicitlyDeclared))
+                {
+                    // One events with add or remove method
+                    editor.SetAccessibility(declaration, Accessibility.Public);
+                    editor.SetModifiers(declaration, modifiers);
+                    return;
+                }
+                else
+                {
+                    // Several events are declared same line
+                    var publicAndNonStaticSymbol = CodeGenerationSymbolFactory.CreateEventSymbol(
+                        eventSymbol,
+                        accessibility: Accessibility.Public,
+                        modifiers: modifiers);
+                    var options = new CodeGenerationOptions(generateMethodBodies: true);
```

Comment:
why do you need 'generateMethodBodies'? (doc in the code)

Reply:
This value value doesn't really matter, I will change it to false since it seems like more meaningful since event don't have method body